File: manual/src/test/java/examples/java/CustomExtensionsDocTest.java
package examples.java;

import static org.approvej.ApprovalBuilder.approve;
import static org.approvej.scrub.Replacements.labeled;

import org.junit.jupiter.api.Test;

@org.approvej.ApprovalTest
class CustomExtensionsDocTest {

  @Test
  void custom_scrubber() {
    // tag::custom_scrubber[]
    approve("Contact jane@example.com or john@company.org for details.")
        .scrubbedOf(new EmailScrubber()) // <1>
        .byFile();
    // end::custom_scrubber[]
  }

  @Test
  void custom_scrubber_replacement() {
    // tag::custom_scrubber_replacement[]
    approve("Contact jane@example.com or john@company.org for details.")
        .scrubbedOf(new EmailScrubber().replacement(labeled("redacted"))) // <1>
        .byFile();
    // end::custom_scrubber_replacement[]
  }
}

File: manual/src/test/kotlin/examples/kotlin/CustomExtensionsDocTest.kt
package examples.kotlin

import org.approvej.ApprovalBuilder.approve
import org.approvej.scrub.Replacements.labeled
import org.junit.jupiter.api.Test

@org.approvej.ApprovalTest
class CustomExtensionsDocTest {

  @Test
  fun `custom scrubber`() {
    // tag::custom_scrubber[]
    approve("Contact jane@example.com or john@company.org for details.")
      .scrubbedOf(EmailScrubber()) // <1>
      .byFile()
    // end::custom_scrubber[]
  }

  @Test
  fun `custom scrubber replacement`() {
    // tag::custom_scrubber_replacement[]
    approve("Contact jane@example.com or john@company.org for details.")
      .scrubbedOf(EmailScrubber().replacement(labeled("redacted"))) // <1>
      .byFile()
    // end::custom_scrubber_replacement[]
  }
}
